### PR TITLE
EXAMPLES: Add recv callback for receiving FIN message

### PR DESCRIPTION
**Pull request by ivankochin**
## What
Defines AM receive callback on the client side. Additionally extends the CI to cover this case.

## Why ?
This callback is required in case of using RNDV proto to get FIN msg.

### CODE REVIEW

File: examples/ucp_client_server.c [hunk 1]
```diff
@@ -903,6 +917,15 @@ static int client_server_do_work(ucp_worker_h ucp_worker, ucp_ep_h ep,
         }
     }
 
+    /* Register recv callback on the client side to receive FIN message */
+    if ((!is_server) && (send_recv_type == CLIENT_SERVER_SEND_RECV_AM)) {
```
Comment (yosefe) [inline examples/ucp_client_server.c:921]: extra ()

File: examples/ucp_client_server.c [hunk 2]
```diff
@@ -886,10 +886,24 @@ start_server(ucp_worker_h ucp_worker, ucx_server_ctx_t *context,
     return status;
 }
 
+ucs_status_t register_am_recv_callback(ucp_worker_h worker) {
```
Comment (yosefe) [inline examples/ucp_client_server.c:889]: { on next line

Comment (ivankochin) [reply examples/ucp_client_server.c:889]:
```suggestion
ucs_status_t register_am_recv_callback(ucp_worker_h worker)
{
```

File: examples/ucp_client_server.c (hunk 1 above)
Comment (ivankochin) [reply examples/ucp_client_server.c:921]:
```suggestion
    if (!is_server && (send_recv_type == CLIENT_SERVER_SEND_RECV_AM)) {
```

**Review by yosefe — APPROVED** [review]

**Comment by ivankochin** [comment]
@gleon99 pls take a look and approve if you have no comments

**Review by yosefe — APPROVED** [review]

**Pull request merged**